tests/manage/pv_services/test_del_mon_service_and_create_pvc.py

```diff
@@ -235,3 +237,91 @@ def test_pvc_creation_after_del_mon_services(self, interface, pod_factory):
+        5. Create PVC, should succeeded.
+
+        """
+        self.sanity_helpers = Sanity()
+
+        # Get all mon services
+        mon_svc_before = get_services_by_label(
+            label=constants.MON_APP_LABEL,
+            namespace=constants.OPENSHIFT_STORAGE_NAMESPACE,
+        )
+
+        # Get all mon pods
+        mon_pods = get_mon_pods()
+
+        # Delete the mon services one by one
+        svc_obj = OCP(
+            kind=constants.SERVICE, namespace=constants.OPENSHIFT_STORAGE_NAMESPACE
+        )
+        mon_svc_ip_before = []
+        for svc in mon_svc_before:
+            svc_name = svc["metadata"]["name"]
+            mon_svc_ip_before.append(svc["spec"]["clusterIP"])
+            log.info(f"Delete mon service {svc_name}")
+            svc_obj.delete(resource_name=svc_name)
+            # Verify mon services deleted
+            svc_obj.wait_for_delete(resource_name=svc_name)
+
+        # Restart the rook-operator pod
+        operator_pod_obj = get_operator_pods()
+        delete_pods(pod_objs=operator_pod_obj)
+        POD_OBJ.wait_for_resource(
+            condition=constants.STATUS_RUNNING, selector=constants.OPERATOR_LABEL
+        )
+
+        # Verify same mon services are created again
+        for svc in mon_svc_before:
+            svc_name = svc["metadata"]["name"]
+            svc_obj.check_resource_existence(
+                should_exist=True, timeout=300, resource_name=svc_name
+            )
+        log.info("Same old mon services are recreated")
+
+        # Validate all mons are running
+        log.info("Validate all mons are up and running")
+        POD_OBJ.wait_for_resource(
+            condition=constants.STATUS_RUNNING,
+            selector=constants.MON_APP_LABEL,
+            resource_count=len(mon_pods),
+            timeout=600,
+            sleep=3,
+        )
+
+        # Validate same mon services are running
+        log.info("Validate same mon services are running")
+        mon_svc_after = get_services_by_label(
+            label=constants.MON_APP_LABEL,
+            namespace=constants.OPENSHIFT_STORAGE_NAMESPACE,
+        )
+        mon_svc_ip_after = [svc["spec"]["clusterIP"] for svc in mon_svc_after]
+        assert len(set(mon_svc_ip_after) ^ set(mon_svc_ip_before)) == 0, (
+            "Different mon services are running. "
+            f"Before mon services list: {mon_svc_ip_before}, "
+            f"After mon services list: {mon_svc_ip_after}"
+        )
+        log.info("Same old mon services are running and all mons are in running state")
+
+        # Verify everything running fine
+        log.info("Verifying All resources are Running and matches expected result")
+        self.sanity_helpers.health_check(tries=120)
+
+        # Validate all storage pods are running
+        wait_for_storage_pods()
+
+        # Create and delete resources
+        self.Sanity.create_pvc_delete(multi_pvc_factory=multi_pvc_factory)
```

Comment:
https://bugzilla.redhat.com/show_bug.cgi?id=1858195#c52 mentions "After the restart of mgr pod, picked the new mon endpionts.". if this is not done then will it not impact the next set of tests in the suite during execution?

Reply:
This test case verifies that the same mon services come up and join the quorum after deleting them manually; new mon services/endpoints are not created so we no need to restart the mgr pod and verify. The fix for the bug is that suppose if on mon failure same mon services/endpoints should come up https://bugzilla.redhat.com/show_bug.cgi?id=1969733#c27